src/coreclr/gc/unix/gcenv.unix.cpp

```diff
@@ -168,6 +168,17 @@ enum membarrier_cmd
 
 bool CanFlushUsingMembarrier()
 {
+
+#ifdef TARGET_ANDROID
+    // Avoid calling membarrier on older Android versions where membarrier
+    // may be barred by seccomp causing the process to be killed.
+    int apiLevel = android_get_device_api_level();
+    if (apiLevel < __ANDROID_API_Q__)
```

Comment:
Maybe check if `seccomp` is enabled for the process first to unblock the usage on older versions?

Reply:
ART doesn't bother with these old versions either. I only see it as more opportunities for bugs. We already have a problem testing _this_ fix.